### PR TITLE
Create Table: check_type #59

### LINKED ISSUE #59: Create Table: check_type
### Overview

We need to create the check_type table so that we can update a shared data store across hackforla.org, vrms, civictechjobs, and tables (onboarding) project.

#### Details

A table and a model are the same thing

### Action Items

- [ ] identify and document table description (see spreadsheet under Resources)
  - [ ] if not, reach out to PD leads
- [ ] compare and check off the data fields (below) against the ERD. Note any fields in the ERD that's missing from the list. (see Resources)
- [ ] compare the check off associated tables (below) against the ERD. Note any associated tables in the ERD that's missing from the list (see Resources)
- [ ] create a single model in Django (defining schema)
- [ ] write a test for the relationships this model will have with other models (e.g., creating a user and assigning them a set of permissions on a project).
- [ ] write an API end point
- [ ] write API unit tests
- [ ] document the endpoint

### Resources/Instructions

- See [People Depot Resources wiki page](https://github.com/hackforla/peopledepot/wiki/Resources-and-Links) for links
  - ERD
  - Table and Field Definitions
  - API Endpoint
- See the [wiki page](https://github.com/hackforla/peopledepot/wiki/Create-Table-issues-data-gathering-workflow) for instructions.

### Description

Types of checks we perform

### Data Fields

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)

    - [ ] (PK) id - int
    - [ ] name - varchar
    - [ ] description - varchar

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

### Associated Tables

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)
   - [ ] user_check (one-to-many)

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

### PR DESCRIPTION
**Pull request by dmartin4820**
Fixes #59 

### What changes did you make?
- create a single model in Django (defining schema)
- add seed data
-write a test for the relationships this model will have with other models (e.g., creating a user and assigning them a set of
- permissions on a project).
- write an API end point
- write API unit tests
- CONTRIBUTING.md: add relative path starting from ./app to step 2.3.2
- add-model-and-api-endpoints.md: add import model to views.py step 

### Why did you make the changes (we will use this info to test)?
- create table for check_type, completing one task as part of #25 
- updated docs to clarify and/or include steps I didn't find in the docs

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals change</summary>

- add-model-and-api-endpoints.md before:
add_model_before

- add-model-and-api-endpoints.md after:
add_model_after

- CONTRIBUTING.md before:
contributing_before

- CONTRIBUTING.md after:
contributing_after

</details>

### CODE REVIEW

**Review by fyliu — CHANGES_REQUESTED** [review]
Thanks for working through the issue so quickly and fixing documentation problems!

Everything works great. I looked through the code, built it, ran it, checked migrations, the admin site, and the API endpoints.

I found only very minor things that doesn't affect the functionality. Can you please fix them? Thanks!

File: app/core/admin.py
Comment (dmartin4820) [reply app/core/admin.py]: Changed in commit 89ce240

File: docs/how-to/add-model-and-api-endpoints.md
Comment (dmartin4820) [reply docs/how-to/add-model-and-api-endpoints.md]: Changed in commit 89ce240

**Review by fyliu — APPROVED** [review]
Thank you for making the changes. Looks great!

**Comment by fyliu** [comment]
@dmartin4820 I'm going to amend the later changes into the original commits before rebasing everything into main. I'm not sure if it's something you want to try doing yourself.

Also, the repo is set to delete the PR branch after merging. Since you're using the main branch of your from for the PR, I'm not sure if it's aware enough to not try to delete it. The delete is supposed to be reversible if it does happen.

**Comment by dmartin4820** [comment]
@fyliu sure, I'd be willing to try amending the commits. I'm familiar with soft resetting to a specific commit, the amending the unstaged changes to the last commit. After that, I'd need to force push to my fork

Not sure I know how to fixup the unstaged changes to specific commits, especially without a tool. But willing to learn more about it. 

I don't mind if it deletes the branch of the fork or whatever it ends up doing. I can create a new one.

**Comment by fyliu** [comment]
Great. First, split the last commit into 2. You can soft reset 2 commits back so that all of the Fixup changes are there together. Actually, mixed reset might be better in this case. Mixed will unstage the committed changes while soft reset will keep them staged.

Then stage one of the changes like the app/core/admin.py change that could become the fixup for the "...register admin..." commit.
So have just that one change staged, then do 'git commit --fixup :/"register admin"' and it will create a fixup commit for the last commit in history with a short msg that matches that partial string. You can try doing the fixup now by running git commit -i --autosquash HEAD^10 . There's only 8-9 commits from HEAD back to "main", so 10 will go back enough to cover all of your changes. It'll open an editor with the rebase plan, and it'll move the fixup commit to right after the commit it's fixing up, with a fixup action in that line. Just close the editor and it will apply it correctly.

Then you can create the fixup commits for the other 2 things and apply them together. I just realized the remaining change is actually one thing, which is fixing the '...import model..." commit. I just wanted to say you can apply multiple fixups in the same autosquash commit.

After that, you should've combined all the fixes into the original commits and still have the exact same code as when the PR was approved. You can check using "git diff origin/main" I think. Then you can force push to replace what's in the PR.

In open source projects, some (many?) maintainers like to keep the main branch history clean and linear. So no fix commit for something earlier in the same PR and no merge commits. So rebase your commits to the latest main if main was changed before your PR.

They might do the cleanup themselves, ask contributors to do it, or have a merge team that's very adept at handling this. 

Thanks for wanting to try this. I wanted to give people this experience that they could encounter in the wild, but I also understand some people are at the stage where they're already learning several things at the same time and don't want another complication.

**Comment by fyliu** [comment]
I can't remember the way to do it with a tool (lazygit) without using the computer, so I'll tell you later. That tool might do several git commands in the background for every action you tell it to take. You can do things like browse an old commit and tell it to commit part of it as a separate commit, splitting it into 2 at that point in history, with all the later commits after them.

**Comment by fyliu** [comment]
## lazygit instructions

I did 2 things:

1. moved the pre-commit commit to before the "docs(add-model)" commit and applied it as a fixup.
1. split the last commit into 2 and moved and applied them to the correct commits in history.

### move and apply fixup

1. assume you're running `lazygit` and on the correct branch
1. focus on the commit panel by pressing "4"
1. move focus to the "pre-commit" line by pressing "j" and "k" to move the cursor down and up
1. move the commit down a position by pressing "ctrl-j"
1. apply the fixup by pressing "f" and "enter"

### split a commit into 2

1. assume you're running `lazygit` and on the correct branch
1. focus on the commit panel by pressing "4"
1. move focus to the commit you want to split by pressing "j" and "k" to move down and up. in this case, the top commit is where we want to be
1. explore the commit by pressing "enter"
1. stage the `admin.py` file by focusing on it and pressing "space"
1. call up the patch menu by pressing "ctrl-p"
1. select "Move patch into new commit" or press "n"
1. enter a new commit message like "fix admin class name" and press "enter"
1. the operation may take a number of seconds.

### move and apply fixup

same steps as above with the 2 newly split commits

**Comment by dmartin4820** [comment]
I'll give it a try. Definitely not opposed to learning more about Git

**Comment by dmartin4820** [comment]
So I did the fixups using lazygit. It was easier that way. It got rid of the "pre-commit" commit and fixedup the commits with the changes you suggest to the admin class name and docs.

**Comment by fyliu** [comment]
@dmartin4820 looks great. Thanks!

**Pull request merged**
